cmd/jimmsrv/main.go

```diff
@@ -207,30 +207,36 @@ func start(ctx context.Context, s *service.Service) error {
 		Handler:           jimmsvc,
 		ReadHeaderTimeout: time.Second * 5,
 	}
+
+	maxConccurentConncetions, _ := strconv.Atoi(os.Getenv("JIMM_SSH_MAX_CONCURRENT_CONNECTIONS"))
+
+	sshServer, err := ssh.NewJumpServer(ctx, ssh.Config{
+		Port:                     os.Getenv("JIMM_SSH_PORT"),
+		HostKey:                  []byte(hostKeyRaw),
+		MaxConcurrentConnections: maxConccurentConncetions,
+	}, jimmsvc.JIMM().SSHManager())
+
 	s.OnShutdown(func() {
 		ctx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
 		defer cancel()
 
-		zapctx.Warn(ctx, "server shutdown triggered")
+		zapctx.Warn(ctx, "HTTP server shutdown triggered")
 		err = httpsrv.Shutdown(ctx)
 		if err != nil {
-			zapctx.Error(ctx, "failed to shutdown server gracefully", zap.Error(err))
+			zapctx.Error(ctx, "failed to shutdown HTTP server gracefully", zap.Error(err))
```

Comment:
shutdown implies graceful, can just say "failed to shutdown HTTP server" and same for SSH

Reply:
No? Shutdown can be forced or graceful. In this case it wasn't graceful, something went wrong during shutdown.

Reply:
Usually the language "shutdown" just implies graceful? Terminated / Kill etc often are ungraceful? And it still attempted it gracefully?

Reply:
No, inside the function, if the shutdown didn't complete by the time the context was cancelled then it returns an error to indicate shutdown didn't happen within the alloted time.

Reply:
Yup, if it was ungraceful, you wouldn't have any timer - it'd be "burn this thing with fire now!!"